type_inference/type_retrieval_service.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/python
+#
+# Copyright 2023 Logica Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict
+from type_inference.type_retrieval_exception import TypeRetrievalException
+from type_inference.type_retriever import TypeRetriever
+import psycopg2
+
+
+def ValidateRuleAndGetTableName(rule: dict) -> str:
+  rule_text = rule['full_text']
+  field_value = rule['head']['record']['field_value']
+
+  if len(field_value) != 1 or field_value[0]['field'] != '*':
+    raise TypeRetrievalException(rule_text)
+
+  conjuncts = rule['body']['conjunction']['conjunct']
+
+  if len(conjuncts) != 1:
+    raise TypeRetrievalException(rule_text)
+
+  conjunct = conjuncts[0]
+
+  if 'predicate' not in conjunct:
+    raise TypeRetrievalException(rule_text)
+
+  field_values = conjunct['predicate']['record']['field_value']
+
+  if len(field_values) != 1 or field_values[0]['field'] != '*':
+    raise TypeRetrievalException(rule_text)
+
+  return conjuncts[0]['predicate']['predicate_name'].split('.')[1]  # TODO: Validate schema of the called table.
+
+
+class TypeRetrievalService:
```

Comment:
Should we call it PostgresTypeRetrievalService? Or other engines would be coming to this class too?

Reply:
No, I suppose that we should make another services for every engine